experimental/algorithm/LAGraph_CFG_reach_basic.c

```diff
@@ -0,0 +1,159 @@
+#define LG_FREE_WORK                                                                     \
+    do {                                                                                 \
+        for (size_t i = 0; i < T_size; i++) {                                            \
+            GrB_free(&T[i]);                                                             \
+        }                                                                                \
+        free(nnz);                                                                       \
+    } while (0)
+
+#define LG_FREE_ALL                                                                      \
+    { LG_FREE_WORK; }
+
+#include "LG_internal.h"
+#include <LAGraphX.h>
+
+// Determines the set of vertex pairs (u, v) in graph (which represented by adjacency
+// matrices) such that there is a path from u to v whose edge labels form a word from the
+// language generated by the context-free grammar (which represented by `rules`).
+//
+// Terminals and non-terminals enumerated by integers starting with zero.
+GrB_Info LAGraph_CFG_reach_basic(
+    // output
+    GrB_Matrix *output, // Matrix with result. (i, j) = true <=> there is a path from i to
```

Comment:
I suppose, it would be better to return matrices for all neterminals. Additionally we can provide a wrapper that filters all except start nonterminal out.